app/views/forumthreads/search.html.erb

```diff
@@ -0,0 +1,56 @@
+<% title "Thread Search" %>
+<h1>Thread Search</h1>
+<h3>Leave a field blank to ignore that search aspect.</h3>
+<% label = Label.where(name: params[:label]).first %>
+<table>
+  <tbody>
+<%= form_tag({controller: "forumthreads", action: "search_redirect"}, method: :post) do %>
+  <%
+    forums = []
+    Forum.all.sort_by{ |f| f.forumgroup && f.forumgroup.position || 0 }.each do |f|
+      if current_user != nil && current_user.role_id > f.role_read_id.to_i || current_user == nil && f.role_read_id == nil
```

Comment:
Are you trying to do permission checks? Please use `f.can_read?` instead.
Also note: Roles are ranked by their value, not id (!). And a forum's `role_read` can be nil while its `role_read_id` is not. (e.g. role deleted)